sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SpanDescriptionExtractor.java

```diff
@@ -0,0 +1,62 @@
+package io.sentry.opentelemetry;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import io.sentry.protocol.TransactionNameSource;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class SpanDescriptionExtractor {
+
+  public @NotNull SpanDescription extractSpanDescription(final @NotNull ReadableSpan otelSpan) {
+    final @NotNull String name = otelSpan.getName();
+
+    final @Nullable String httpMethod = otelSpan.getAttribute(SemanticAttributes.HTTP_METHOD);
+    if (httpMethod != null) {
+      return descriptionForHttpMethod(otelSpan, httpMethod);
+    }
+
+    final @Nullable String dbSystem = otelSpan.getAttribute(SemanticAttributes.DB_SYSTEM);
+    if (dbSystem != null) {
+      return descriptionForDbSystem(otelSpan);
+    }
+
+    return new SpanDescription(name, name, TransactionNameSource.CUSTOM);
+  }
+
+  private SpanDescription descriptionForHttpMethod(
+      final @NotNull ReadableSpan otelSpan, final @NotNull String httpMethod) {
+    final @NotNull String name = otelSpan.getName();
+    final @NotNull SpanKind kind = otelSpan.getKind();
+    final @NotNull StringBuilder opBuilder = new StringBuilder("http");
+
+    if (SpanKind.CLIENT.equals(kind)) {
+      opBuilder.append(".client");
+    } else if (SpanKind.SERVER.equals(kind)) {
+      opBuilder.append(".server");
+    }
+    final @Nullable String httpTarget = otelSpan.getAttribute(SemanticAttributes.HTTP_TARGET);
+    final @Nullable String httpRoute = otelSpan.getAttribute(SemanticAttributes.HTTP_ROUTE);
+    final @Nullable String httpPath = httpRoute != null ? httpRoute : httpTarget;
```

Comment:
shouldn't it be just `http.client/http.server` according to https://develop.sentry.dev/sdk/performance/span-operations/? Or we're changing this?

Reply:
Hmm, `opBuilder` should create any of `http`, `http.client` or `http.server` so exactly what's asked for, no?

Reply:
ah, sorry, I thought you're adding it to the opBuilder as well, missed that